tests/publiccloud/ssh_interactive_start.pm

```diff
@@ -16,17 +16,40 @@ use Mojo::Base 'publiccloud::ssh_interactive_init';
 use publiccloud::ssh_interactive;
 use testapi;
 use utils;
+use publiccloud::utils "select_host_console";
 
 sub run {
     my ($self, $args) = @_;
-    select_console 'tunnel-console';
+
+    # This ensure that we have used the setup console, even if no module was run before.
+    select_host_console();
+    my $setup_console = current_console();
 
     # Establish the tunnel (it will stay active in foreground and occupy this console!)
+    select_console('tunnel-console');
     ssh_interactive_tunnel($args->{my_instance});
 
-    # Switch to root-console and SSH to the instance
-    # every other loaded test must stay in root-console
-    select_console 'root-console';
+    # Enable ssh connection on setup console, this is done normally with the
+    # first activation hook in susedistribution:activate_console()
+    if ($setup_console !~ /tunnel/) {
+        select_console($setup_console);
+        script_run('ssh -t sut', timeout => 0);
+    }
+
+    die("expect ssh serial") unless (get_var('SERIALDEV') =~ /ssh/);
+
+    # Verify most important consoles
+    select_console('root-console');
+    assert_script_run('test -e /dev/' . get_var('SERIALDEV'));
+    assert_script_run('test $(id -un) == "root"');
+
+    select_console('user-console');
+    assert_script_run('test -e /dev/' . get_var('SERIALDEV'));
+    assert_script_run('test $(id -un) == "' . $testapi::username . '"');
+
+    $self->select_serial_terminal();
+    assert_script_run('test -e /dev/' . get_var('SERIALDEV'));
+    assert_script_run('test $(id -un) == "root"');
```

Comment:
I would move this to `tests/publiccloud/instance_overview.pm`.

Reply:
It's not related to the instance. The purpose of these lines are:

Check on `root-console`, `user-console` and `serial_teriminal (e.g. `root-virtio-terminal` ) that the ssh tunnel is setup properly.
Actually this is just a `sanity check for this `TUNNELED` behavior. We could remove it, but I found it handy so we are aware that it worked once.

Reply:
The `instance_overview` test module is also basically just sanity checking :)